### ISSUE TITLE
Accuracy for multi-output model is not working 

### ISSUE THREAD

**Issue by GKalliatakis** [keras-team/keras#10287]
An important concern in the case of multi-output models is that the training of such a model requires the ability to specify different metrics for different heads (outputs) of the network.

As mentioned in the official documentation: 
> To specify different metrics for different outputs of a multi-output model, you could also pass a dictionary, such as metrics={'output_a': 'accuracy'}

For my model, I am doing something similar with the following :
```python
    metrics ={'output_a': 'crossentropy',
              'output_b': 'mse',
              'output_c': 'mse',
              'output_d': 'mse',
              'output_e': 'categorical_accuracy'}
```

**But when I start training the model, the overall accuracy is nowhere to be seen at the logs, while the `loss` and the `val_loss` are visible.**

So my questions are:

1. Do `val` and `val_loss` imply the overall loss and overall validation loss of the model respectively? 
2. Is it possible to have the `acc` of the model printed as well? 

More info about the model can be found at #10120, while a similar issue was discussed at #2412 without clear solution.

**Comment by Liu-Da** [comment]
same question

**Comment by GKalliatakis** [comment]
So from my experience even though I was not able to find clear answers I can answer the following to my own questions:
1. `val` and `val_loss` **DO** imply the overall loss and overall validation loss of the model respectively. The individual losses for each output are weighted according to the coefficients specified in `loss_weights`
2. You can have the accuracy for each output individually, but Keras doesn't support **_overall metrics_**. This would require more information on how the individual outputs' metrics should be aggregated.

Refer to this answer as well [https://stackoverflow.com/questions/50541409/keras-accuracy-for-multi-output-model-is-not-working](https://stackoverflow.com/questions/50541409/keras-accuracy-for-multi-output-model-is-not-working )

**Comment by Liu-Da** [comment]
Can anyone give some suggestions or provide some sample code?

**Comment by QuickLearner171998** [comment]
I have a 13 output model. I needed to monitor the mean val accuracy of all outputs. Creating a Custom callback works for me. I use ```on_epoch_end``` to calculate the mean accuracy and then save the model with a specific name (having epoch and val accuracy). Also with the help of the model name, I keep a track of the best val accuracy so far.
These are some sample codes - 
```
def get_val_acc_from_file(save_path):
  return float(save_path.split('_')[-1][:-3])


def return_best_saved_model(paths):
  val_acc = 0
  ret_path = ""
  for path in paths:
    best_val_acc = get_val_acc_from_file(path)
    if(best_val_acc >= val_acc):
      val_acc = best_val_acc
      ret_path = path
  return ret_path

class CustomCallbacks(tf.keras.callbacks.Callback):
  def on_epoch_end(self, epoch, logs={}):
    val_acc = 0
    for i in range(13):
      val_acc += (logs.get('val_digit_{}_accuracy'.format(i)))/13 # take mean acc value
    if len(glob.glob("./*.h5")):
      # name format wt_epoch_valacc.h5
      best_model_path = return_best_saved_model(glob.glob("./*h5"))
      prev_best_val_acc = get_val_acc_from_file(best_model_path)
      if(val_acc > prev_best_val_acc):
        print("\nSaving Model...\n")
        self.model.save('wt_{}_{}.h5'.format(epoch, round(val_acc,4)))
    else:
      self.model.save('wt_{}_{}.h5'.format(epoch, round(val_acc, 4)))
```

PS - I know the functions can be improved further but these are just to give the essence.
